src/net.h

```diff
@@ -1408,6 +1408,11 @@ class CConnman
     uint16_t GetDefaultPort(Network net) const;
     uint16_t GetDefaultPort(const std::string& addr) const;
 
+    /**
+     * Add entries to `m_unused_i2p_sessions` if needed.
+     */
+    void AddI2PSessionsIfNeeded() EXCLUSIVE_LOCKS_REQUIRED(!m_unused_i2p_sessions_mutex);
+
     // Network usage totals
```

Comment:
The `AddI2PSessionsIfNeeded` function is declared with `EXCLUSIVE_LOCKS_REQUIRED(!m_unused_i2p_sessions_mutex)`, which is incorrect. It should require the lock, not the absence of it.